File: src/IsbInterop/ComponentTokens/IComponentTokenFactory.vb
Imports IsbInterop.Base

Namespace ComponentTokens

  ''' <summary>
  ''' Фабрика задач.
  ''' </summary>
  Public Interface IComponentTokenFactory
    Inherits IEdmsObjectFactory(Of IComponentToken, IComponentTokenInfo)

    ''' <summary>
    ''' Запускает компоненту.
    ''' </summary>
    ''' <param name="objectInfo">Информация о варианте запуска компоненты.</param>
    Sub Execute(objectInfo As IObjectInfo)

    ''' <summary>
    ''' Запускает компоненту в новом процессе.
    ''' </summary>
    ''' <param name="objectInfo">Информация о варианте запуска компоненты.</param>
    Sub ExecuteInNewProcess(objectInfo As IObjectInfo)
  End Interface
End Namespace

File: src/IsbInterop/ComponentTokens/IComponentTokenFactory.cs
using IsbInterop.Base;

namespace IsbInterop.ComponentTokens
{
  /// <summary>
  /// Фабрика задач.
  /// </summary>
  public interface IComponentTokenFactory : IEdmsObjectFactory<IComponentToken, IComponentTokenInfo>
  {
    /// <summary>
    /// Запускает компоненту.
    /// </summary>
    /// <param name="objectInfo">Информация о варианте запуска компоненты.</param>
    void Execute(IObjectInfo objectInfo);

    /// <summary>
    /// Запускает компоненту в новом процессе.
    /// </summary>
    /// <param name="objectInfo">Информация о варианте запуска компоненты.</param>
    void ExecuteInNewProcess(IObjectInfo objectInfo);
  }
}
